clash-cores/LICENSE

```diff
@@ -1,4 +1,5 @@
 Copyright (c) 2019, Foamspace Corp.
+              2023, Amaranth HDL contributors
```

Comment:
Parts of the implementation are based on their implementation. We should comply with their license. I didn't feel the need to add it to the source code files since the code is written from scratch. If you prefer this I can do that as well.

Reply:
I think maybe it's better to put them in the source file and explain why they are there. The current solution _does_ mention they have a copyright here, but it's not really clear why. I'm not sure about the legal rules, but it seems fair to mention explicitly that they cooked up part of the CRC solution specifically. And I think it would not be weird to put that in `CRC/Internal.hs` in any way.